### PR TITLE
docs: add Category description & constraints

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

I add a description and constraints of `Category`.

## Changes

- Add a Category description
- Add a Category constraints

> ## Category - カテゴリー
> 
> ### Category ID
> 
> Category ID is an unique ID.
> 
> ### Name
> 
> The name of group to which the task belongs.  
> Kitchen, Bathroom, Entrance and Living room are set by default. The landlord can create additional categories as they like apart from those listed.
> 
> #### Constraints
> 
> - the name must be unique.
> - It must be less than equal to 15 letters and greater than equal to 1.
> - The maximum number of categories per sharehouse is 20.
> 
> ### Task
> 
> Category contains Task. Category has Task list.
> 

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)

## Reviewer Checklist

- Code readability and simplicity
- Follows best practices and coding standards
- Understandable and maintainable code

### CODE REVIEW

File: docs/docs/modeling.md [hunk 1]
```diff
@@ -108,6 +107,25 @@ classDiagram
...
+### Name
+
+The name of group to which the task belongs.  
+Kitchen, Bathroom, Entrance and Living room are set by default. The landlord can create additional categories as they like apart from those listed.
+
+#### Constraints
+
+- the name must be unique.
+- It must be less than equal to 15 letters and greater than equal to 1.
+- The maximum number of categories per sharehouse is 20.
```
Comment (akiomatic) [inline docs/docs/modeling.md:123]: I think the following constraint belongs to the `Category` itself, not just to the `Name`.
- The maximum number of categories per sharehouse is 20.

File: docs/docs/modeling.md [hunk 2]
```diff
@@ -108,6 +107,25 @@ classDiagram
 
 ## Category - カテゴリー
```
Comment (akiomatic) [inline docs/docs/modeling.md:108]: I'd like you to describe what the `Category` is here.

Comment (akiomatic) [inline docs/docs/modeling.md:108]: You might need to mention the minimal number of `Category` too as well as the maximum you mentioned below.

Comment (akiomatic) [inline docs/docs/modeling.md:108]: Should we make the maximum number of `Task` that a `Category` can have?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you!
I kindly ask you to check out my comments!

File: docs/docs/modeling.md (hunk 2 above)
Comment (KaiKoide) [reply docs/docs/modeling.md:108]: Thank you for your advice. We need to talk about that.

**Comment by KaiKoide** [comment]
I have modified the descriptions in the mentioned parts.

**Review by akiomatic — APPROVED** [review]
LGTM!
I apologize for the late review.

**Review by HrRn — APPROVED** [review]
Thank you!

**Review by yuki-92c — APPROVED** [review]
Thanks. LGTM

**Review by Akikaze1119 — APPROVED** [review]
Thank you! LGTM👍

**Pull request merged**